From bd0d9d4ac9148211e14eb1548ad54829af465aed Mon Sep 17 00:00:00 2001
From: Chris Halls <halls@debian.org>
Date: Mon, 24 Aug 2009 16:44:15 +0100
Subject: [PATCH] Prevent multiple window resize calls when maximised

See http://bugs.debian.org/542387.

Problem introduced as part of fix to IZ#44325.

If KDE Window manager receives a window move after a window has already been
maximised, it assumes the window should be restored to original size. Under
Gnome it will stay maximised.

This fix disables further window resize calls once the window state has been
set to maximised.
---
 unx/gtk/window/gtkframe.cxx |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git vcl/unx/gtk/window/gtkframe.cxx b/unx/gtk/window/gtkframe.cxx
index c1fbbc5..d5e514a 100644
--- vcl.old/unx/gtk/window/gtkframe.cxx
+++ vcl/unx/gtk/window/gtkframe.cxx
@@ -1501,7 +1501,7 @@ void GtkSalFrame::SetPosSize( long nX, long nY, long nWidth, long nHeight, USHOR
 
         if( isChild( false, true ) )
             gtk_widget_set_size_request( m_pWindow, nWidth, nHeight );
-        else
+        else if( ! ( m_nState & GDK_WINDOW_STATE_MAXIMIZED ) )
             gtk_window_resize( GTK_WINDOW(m_pWindow), nWidth, nHeight );
         setMinMaxSize();
 	}
@@ -1603,6 +1603,7 @@ void GtkSalFrame::SetWindowState( const SalFrameState* pState )
         SAL_FRAMESTATE_MASK_MAXIMIZED_WIDTH | SAL_FRAMESTATE_MASK_MAXIMIZED_HEIGHT;
     
     if( (pState->mnMask & SAL_FRAMESTATE_MASK_STATE) &&
+        ! ( m_nState & GDK_WINDOW_STATE_MAXIMIZED ) &&
         (pState->mnState & SAL_FRAMESTATE_MAXIMIZED) &&
         (pState->mnMask & nMaxGeometryMask) == nMaxGeometryMask )
     {
-- 
1.6.3.3

